### PR TITLE
[GEOT-7014] Move functionality for filtering tiles by the layer's tileMatrixLimits

### PR DESCRIPTION
**Pull request by roarbra**
[GEOT-7014](https://osgeo-org.atlassian.net/browse/GEOT-7014) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

Some functionality are moved from AbstractGetTileRequest to WMTSTileService. The functionality should be the same, but with less code, and a call for WMTSTileService.findTilesInExtent would potentially return tiles that wasn't within the limits of WMTSLayer.

Some functionality are moved from WMTSTileFactory to WMTSTileService. Leaving behind some deprecated functions.

I've added some tests that covers the change.

# Checklist

- [X] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [X] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [X] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [X] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [X] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [X] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [X] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [X] Bug fixes and small new features are presented as a single commit.
- [X] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/extension/wmts/src/main/java/org/geotools/ows/wmts/client/WMTSTileFactory.java [hunk 1]
```diff
@@ -69,13 +69,15 @@ public Tile findTileAtCoordinate(
     }
 
     /** Find the first valid Upper Left tile for the current layer. */
+    @Deprecated
```
Comment (aaime) [inline modules/extension/wmts/src/main/java/org/geotools/ows/wmts/client/WMTSTileFactory.java:72]: So these public methods are gone, and there is no replacement? I don't think they are widely used, but still... would be nice if there was a way to get the same functionality after the refactor. If there is, the javadoc should say what it is (and the upgrade rst file too).

Comment (roarbra) [reply modules/extension/wmts/src/main/java/org/geotools/ows/wmts/client/WMTSTileFactory.java]: Sorry, I was a little quick there. I will take the deprecation with its own issue at a later stage.

Comment (roarbra) [reply modules/extension/wmts/src/main/java/org/geotools/ows/wmts/client/WMTSTileFactory.java]: @aaime You've changed the status of the ticket in JIRA to Solved, but the PR isn't committed.

Comment (aaime) [reply modules/extension/wmts/src/main/java/org/geotools/ows/wmts/client/WMTSTileFactory.java]: Whoops... easily fixed.

**Pull request merged**